tools/codegen/gen.py

```diff
@@ -29,7 +29,7 @@
 from tools.codegen.utils import (
     Target, concatMap, context, mapMaybe, YamlDumper, YamlLoader, FileManager, assert_never
 )
-from tools.codegen.gen_vmap_plumbing import gen_all_vmap_plumbing
+from tools.codegen.gen_vmap_plumbing2 import gen_all_vmap_plumbing
```

Comment:
probably just missing something - why do we have a `gen_vmap_plumbing2`? And why are we checking the codegen output into version control?

Reply:
> why do we have a gen_vmap_plumbing2? 

I created a new file called gen_vmap_pluming2 to make this PR look nicer, so that one can see the new file with all the codegen instead of a diff view between the old codegen and the new updated codegen. After this PR gets approved I'll apply all the changes to gen_vmap_plumbing instead and remove gen_vmap_plumbing2 (but I can also do that now if you'd prefer).

> And why are we checking the codegen output into version control?

We never had time to set up automatic code generation :P, historically we have [checked in the codegen output into version control](https://github.com/pytorch/functorch/blob/main/functorch%2Fcsrc%2FVmapGeneratedPlumbing.h) and it has worked well enough. There are many reasons for this, but the most important one is that we would like to be able to (1) install a PyTorch nightly binary and (2) build functorch. Adding a codegen step into the functorch build would involve downloading a copy of native_functions.yaml from pytorch which then makes it more difficult for people who are not us to install functorch.

I would really like to get https://github.com/pytorch/pytorch/issues/73212 resolved to make things better

Reply:
That all sounds reasonable! And no preference on when you apply the changes to `gen_vmap_plumbing`.